runtime/v2/manager.go

```diff
@@ -213,186 +276,125 @@ func (m *TaskManager) startShim(ctx context.Context, bundle *Bundle, id string,
+	proc, err := m.list.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
 
-// Add a runtime task
-func (m *TaskManager) Add(ctx context.Context, task runtime.Task) error {
-	return m.tasks.Add(ctx, task)
+	return proc, nil
 }
 
 // Delete a runtime task
-func (m *TaskManager) Delete(ctx context.Context, id string) (*runtime.Exit, error) {
-	task, err := m.tasks.Get(ctx, id)
+func (m *ShimManager) Delete(ctx context.Context, id string) error {
+	proc, err := m.list.Get(ctx, id)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	shim := task.(*shim)
-	exit, err := shim.delete(ctx, m.tasks.Delete)
-	if err != nil {
-		return nil, err
-	}
+	shimTask := proc.(*shimTask)
+	err = shimTask.shim.delete(ctx)
+	m.list.Delete(ctx, id)
 
-	return exit, err
+	return err
 }
 
-// Tasks lists all tasks
-func (m *TaskManager) Tasks(ctx context.Context, all bool) ([]runtime.Task, error) {
-	return m.tasks.GetAll(ctx, all)
+func parsePlatforms(platformStr []string) ([]ocispec.Platform, error) {
+	p := make([]ocispec.Platform, len(platformStr))
+	for i, v := range platformStr {
+		parsed, err := platforms.Parse(v)
+		if err != nil {
+			return nil, err
+		}
+		p[i] = parsed
+	}
+	return p, nil
 }
 
-func (m *TaskManager) loadExistingTasks(ctx context.Context) error {
-	nsDirs, err := os.ReadDir(m.state)
-	if err != nil {
-		return err
-	}
-	for _, nsd := range nsDirs {
-		if !nsd.IsDir() {
-			continue
-		}
-		ns := nsd.Name()
-		// skip hidden directories
-		if len(ns) > 0 && ns[0] == '.' {
-			continue
-		}
-		log.G(ctx).WithField("namespace", ns).Debug("loading tasks in namespace")
-		if err := m.loadTasks(namespaces.WithNamespace(ctx, ns)); err != nil {
-			log.G(ctx).WithField("namespace", ns).WithError(err).Error("loading tasks in namespace")
-			continue
-		}
-		if err := m.cleanupWorkDirs(namespaces.WithNamespace(ctx, ns)); err != nil {
-			log.G(ctx).WithField("namespace", ns).WithError(err).Error("cleanup working directory in namespace")
-			continue
-		}
+// TaskManager wraps task service client on top of shim manager.
+type TaskManager struct {
+	shims *ShimManager
+}
+
+// NewTaskManager creates a new task manager instance.
+func NewTaskManager(shims *ShimManager) *TaskManager {
+	return &TaskManager{
+		shims: shims,
 	}
-	return nil
 }
 
-func (m *TaskManager) loadTasks(ctx context.Context) error {
-	ns, err := namespaces.NamespaceRequired(ctx)
+// ID of the task manager
+func (m *TaskManager) ID() string {
+	return fmt.Sprintf("%s.%s", plugin.RuntimePluginV2, "task")
+}
+
+// Create launches new shim instance and creates new task
+func (m *TaskManager) Create(ctx context.Context, taskID string, opts runtime.CreateOpts) (runtime.Task, error) {
+	process, err := m.shims.Start(ctx, taskID, opts)
 	if err != nil {
-		return err
+		return nil, errors.Wrap(err, "failed to start shim")
 	}
-	shimDirs, err := os.ReadDir(filepath.Join(m.state, ns))
+
+	shim := process.(*shimTask)
```

Comment:
I see there is a note in the previous `Start` call about this being a temporary wrapper. It might be worthwhile to put a note here as well, I had to go looking through the code to figure out why this type assertion was necessary.